File: cses_solutions/introductory/java/BitStrings.java
import java.util.Scanner;

/**
 * The number of bit strings of length N is 2**N (2 to the power of N) because each bit 
 * can be a 0 or a 1 (2 possibilities per bit, n bits, means 2 times itself n times total possibilities).
 * We compute this number modulo 10**9+7 by just taking the mod after each multiplication.
 */
public class BitStrings {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        long n = scanner.nextInt();
        scanner.close();
        long solution = 1;
        for(int i = 0; i < n; i++) {
            solution = (solution * 2) % 1000000007;
        }
        System.out.println(solution);
    }
}

File: cses_solutions/introductory/cpp/BitStrings.cpp
/*
 * Problem link : https://cses.fi/problemset/task/1617
 * 
 * There are n possible bit places and each place can be either 1 or 2 - so two choices.
 * This tells us that we 2^n possible combinations - 2 multiplied by itself n times.
 * In fact, if you look at the bitStrings of any length n, you will see that these are
 * just the binary representations of numbers from 0 to 2^n-1 - so 2^n in total.
 * To find 2^n we can simply multiply a number by 2 n-times. However this will be very
 * big so we take the modulo 1e9+7 every time we multiply by 2.
 * 
 * Runtime O(n)
 * */
#include "bits/stdc++.h"

using namespace std;

int main(){
  int n;
  scanf("%d", &n);
  int ans = 1;
  for(int i=0; i<n; i++)
    ans = (ans * 2) % (1000000007); //multiply by 2 and take mod 1e9+7
  printf("%d\n", ans); //the newline character '\n' isn't necessary but it is a good practice as some online-judges strictly require it
  
  return 0;
}
